### PR TITLE
Improve TMail documentation

### PR DESCRIPTION
**Pull request by chibenwa**
- Overall goals of the TMail suite.
 - Dedicated feature page
 - Contribute page
 - Project ethic page
 - Join us page
 - Presentation of mobile application
 - Futher refine the TMail backend documentation
 
 Grammar nazis reviews welcome cc @Arsnael . Please to code suggestions...
 
 @hoangdat you might be interested by it too...
 
 
Screenshot from 2021-09-30 11-33-26

Fully responsive...

### CODE REVIEW

File: docs/modules/ROOT/pages/tmail-backend/features/index.adoc [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+== The Encrypted mailbox
+
+image::lock.png[Lock logo]
+
+TMail allows your users to store their emails encrypted in their mailbox, using the **GPG format**
+(link:https://datatracker.ietf.org/doc/html/rfc4880[RFC-4880 OpenPGP Message Format],
+link:https://datatracker.ietf.org/doc/html/rfc3156[RFC-3156 MIME Security with OpenPGP]).
+
+Thus, TMail have **no means** to decrypt your messages once they are stored in the encrypted
```
Comment (Arsnael) [inline docs/modules/ROOT/pages/tmail-backend/features/index.adoc:14]:
```suggestion
Thus, TMail has **no means** to decrypt your messages once they are stored in the encrypted
```

File: docs/modules/ROOT/pages/tmail-backend/features/index.adoc [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+mailbox. So be careful not to lose your private key: you would not be able to read your messages ever again! This also
```
Comment (Arsnael) [inline docs/modules/ROOT/pages/tmail-backend/features/index.adoc:15]:
```suggestion
mailbox. So be careful not to loose your private key: you would not be able to read your messages ever again! This also
```

File: docs/modules/ROOT/pages/tmail-backend/jmap-extensions/index.adoc [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+challenges modern mobile fleet requires, it consumes a lot of bandwith, requires a
+lot of roundtrips which means high latency.
+
+We are not alone to say this! Big players of the field started their own link:https://developers.google.com/gmail/api[proprietary]
+link:https://docs.microsoft.com/en-us/exchange/clients/exchange-activesync/exchange-activesync?view=exchserver-2019[protocols]
+to address IMAP flaws, and inter-operable standard was yet to be found...
+
+This, is where link:https://jmap.io[JMAP] comes to play! JMAP builds on decades of experience,
+and beautifully addresses these challenges using `HTTP` and `JSON` standards. Not only does it make
+applications easier to develop, we also strongly believes it results in an improved experience for
```
Comment (Arsnael) [inline docs/modules/ROOT/pages/tmail-backend/jmap-extensions/index.adoc:21]:
```suggestion
applications easier to develop, we also strongly believe it results in an improved experience for
```

File: docs/modules/ROOT/pages/tmail-backend/jmap-extensions/index.adoc [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+
+== TMail JMAP specifications
+
+=== JMAP extensions for TMail
+
+We introduce custom extensions to the main JMAP protocol in order to achieve our means. These extensions are documented here.
+
+These extensions include:
+
+ - link:jmapGpgExtension.adoc[com:linagora:params:jmap:pgp] which add the capability to read messages of TMail encrypted
```
Comment (Arsnael) [inline docs/modules/ROOT/pages/tmail-backend/jmap-extensions/index.adoc:32]:
```suggestion
 - link:jmapGpgExtension.adoc[com:linagora:params:jmap:pgp] which adds the capability to read messages of TMail encrypted
```

File: docs/modules/ROOT/pages/tmail-backend/jmap-extensions/index.adoc [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+ mailbox.
+ - link:ticketAuthentication.adoc[com:linagora:params:jmap:ws:ticket] allow to generate one time usage tickets in order
```
Comment (Arsnael) [inline docs/modules/ROOT/pages/tmail-backend/jmap-extensions/index.adoc:34]:
```suggestion
 - link:ticketAuthentication.adoc[com:linagora:params:jmap:ws:ticket] allows to generate one time usage tickets in order
```

File: docs/modules/ROOT/pages/tmail-backend/jmap-extensions/index.adoc [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+message reception like moving them to a folder.
+ - link:longLivedTokens.adoc[com:linagora:params:long:lived:token] enables a mechanism to safely authenticate mobile devices.
+
+=== Standard JMAP extensions
+
+We furthermore expose the following JMAP specifications in TMail (implemented in James):
+
+ - link:https://datatracker.ietf.org/doc/html/rfc8620[RFC-8620] JMAP Core defines mechanisms shared by all JMAP
+specifications including download, uploads, event source.
+ - link:https://datatracker.ietf.org/doc/html/rfc8621[RFC-8621] JMAP Mail defines JMAP entities for emails and allow
```
Comment (Arsnael) [inline docs/modules/ROOT/pages/tmail-backend/jmap-extensions/index.adoc:47]:
```suggestion
 - link:https://datatracker.ietf.org/doc/html/rfc8621[RFC-8621] JMAP Mail defines JMAP entities for emails and allows
```

File: docs/modules/ROOT/pages/tmail-backend/jmap-extensions/index.adoc [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+ - link:https://datatracker.ietf.org/doc/html/rfc8887[RFC-8887] allows JMAP to rely on webSockets as a transport layer.
+ - link:https://datatracker.ietf.org/doc/html/rfc9007[RFC-9007] JMAP MDN exposes convenient APIs for parsing, and sending
+Message Delivery Notifications (Read receipts).
+
+=== JMAP extensions from Apache James
+
+Moreover, we do expose some JMAP extensions defined within Apache JAMES:
+
+ - link:https://github.com/apache/james-project/blob/master/server/protocols/jmap-rfc-8621/doc/specs/spec/quotas/quotas.mdown[urn:apache:james:params:jmap:mail:quota]
+JMAP Quota (while waiting for the link:https://datatracker.ietf.org/doc/draft-ietf-jmap-quotas/[IETF draft] to be finalized) enable users to know their quotas for the Email related entities.
```
Comment (Arsnael) [inline docs/modules/ROOT/pages/tmail-backend/jmap-extensions/index.adoc:58]:
```suggestion
JMAP Quota (while waiting for the link:https://datatracker.ietf.org/doc/draft-ietf-jmap-quotas/[IETF draft] to be finalized) enables users to know their quotas for the Email related entities.
```

File: docs/modules/ROOT/pages/tmail-backend/jmap-extensions/index.adoc [hunk 8]
```diff
@@ -0,0 +1,95 @@
...
+
+ - TMail supports link:https://en.wikipedia.org/wiki/Basic_access_authentication[Basic Authentication] defined in
+link:https://datatracker.ietf.org/doc/html/rfc7617[RFC-7617]. Please note that while convenient for testing purpose, this
+authentication mechanism should not be used for production workflow: the credentials are transmitted again and over again,
+should be retained in memory, authentication is challenged for each request...
+ - TMail supports link:https://jwt.io/[JWT tokens] (link:https://tools.ietf.org/html/rfc7519[RFC-7519]]) signed by third party
+software, valid according to the configured JWT public key. If valid, the request is blindly trusted.
+link:https://james.staged.apache.org/james-project/3.7.0/servers/distributed/configure/webadmin.html#_generating_a_jwt_key_pair[Read more]...
+ - link:longLivedTokens.adoc[com:linagora:params:long:lived:token] enables a mechanism to safely authenticate mobile devices.
+ - link:ticketAuthentication.adoc[com:linagora:params:jmap:ws:ticket] allow to generate one time usage tickets in order
```
Comment (Arsnael) [inline docs/modules/ROOT/pages/tmail-backend/jmap-extensions/index.adoc:76]:
```suggestion
 - link:ticketAuthentication.adoc[com:linagora:params:jmap:ws:ticket] allows to generate one time usage tickets in order
```

File: docs/modules/ROOT/pages/tmail-backend/jmap-extensions/jmapFilters.adoc [hunk 9]
```diff
@@ -0,0 +1,150 @@
...
+A `Rule` object has the following properties:
+
+ - **id**: Server-set, `Id`, immutable, the JMAP identifier.
+ - **name**: `String`, the name of the rule, which can be used to hold an explanation regarding the purpose of this rule.
+ - **condition**: `Condition`, when should this rule be applied.
+ - **action**: `Action`, what to do when the rule is applied.
+
+A `Condition` object has the following properties:
+
+- **field**: `String`, the filed of the message to be evaluated. Must be one of `from`, `to`, `cc`, `recipient`, `subject`.
```
Comment (Arsnael) [inline docs/modules/ROOT/pages/tmail-backend/jmap-extensions/jmapFilters.adoc:36]:
```suggestion
- **field**: `String`, the field of the message to be evaluated. Must be one of `from`, `to`, `cc`, `recipient`, `subject`.
```

File: docs/modules/ROOT/pages/tmail-backend/jmap-extensions/longLivedTokens.adoc [hunk 10]
```diff
@@ -0,0 +1,166 @@
+= Long Lived Tokens
+:navtitle: Long lived tokens
+
+The goal of this extension is to enable secure mobile connection. The mobile application uses the
+supplied credentions to generate a `Long Lived token` that will be stored on the device. This `long lived token`
```
Comment (Arsnael) [inline docs/modules/ROOT/pages/tmail-backend/jmap-extensions/longLivedTokens.adoc:5]:
```suggestion
supplied credentials to generate a `Long Lived token` that will be stored on the device. This `long lived token`
```

File: docs/modules/ROOT/pages/tmail-backend/jmap-extensions/longLivedTokens.adoc [hunk 11]
```diff
@@ -0,0 +1,166 @@
...
+can be later used to generate `short leaved tokens` with very short validity but that grants access to main
```
Comment (Arsnael) [inline docs/modules/ROOT/pages/tmail-backend/jmap-extensions/longLivedTokens.adoc:6]:
```suggestion
can be later used to generate `short lived tokens` with very short validity but that grants access to main
```

File: docs/modules/ROOT/pages/tmail-backend/jmap-extensions/longLivedTokens.adoc [hunk 12]
```diff
@@ -0,0 +1,166 @@
...
+JMAP APIs.
+
+== Additions to the capability object
+
+Servers supporting the Long lived token extension needs
```
Comment (Arsnael) [inline docs/modules/ROOT/pages/tmail-backend/jmap-extensions/longLivedTokens.adoc:11]:
```suggestion
Servers supporting the Long lived token extension need
```

File: docs/modules/ROOT/pages/tmail-backend/jmap-extensions/longLivedTokens.adoc [hunk 13]
```diff
@@ -0,0 +1,166 @@
...
+
+....
+com:linagora:params:long:lived:token
+....
+
+The associated object is empty.
+
+== The LongLivedToken object
+
+A LongLivedToken have the following fields:
```
Comment (Arsnael) [inline docs/modules/ROOT/pages/tmail-backend/jmap-extensions/longLivedTokens.adoc:22]:
```suggestion
A LongLivedToken has the following fields:
```

File: docs/modules/ROOT/pages/tmail-backend/jmap-extensions/longLivedTokens.adoc [hunk 14]
```diff
@@ -0,0 +1,166 @@
...
+
+ - **id**: `Id`, server set, immutable, the JMAP identifier of the object.
+ - **deviceId**: `String`, immutable, the device identifier associated with this token.
+ - **token**: `String`, immutable, server set, never return by `LongLivedToken/get` calls,
+it is only returned to the creator of the token as a result of `LongLivedToken/set create`. This token can be
+used as a `Bearer` value for the `Authorization` header to generate short lived tokens.
+
+=== LongLivedToken/get
+
+A generic /get method. Do not return the `token` field of `LongLivedToken`.
```
Comment (Arsnael) [inline docs/modules/ROOT/pages/tmail-backend/jmap-extensions/longLivedTokens.adoc:32]:
```suggestion
A generic /get method. Does not return the `token` field of `LongLivedToken`.
```

File: docs/modules/ROOT/pages/tmail-backend/jmap-extensions/longLivedTokens.adoc [hunk 15]
```diff
@@ -0,0 +1,166 @@
...
+....
+{
+  "token" : "eruverlvrververver",
+  "expiresOn" : "2021-09-29T10:55:49.887Z"
+}
+....
+
+Regarding `Authorization` either long lived tokens bearer authentication is supported or basic authentication.
+
+`deviceId` is compulsory whn using long lived token bearer, and needs to be matching the stored value.
```
Comment (Arsnael) [inline docs/modules/ROOT/pages/tmail-backend/jmap-extensions/longLivedTokens.adoc:125]:
```suggestion
`deviceId` is compulsory when using long lived token bearer, and needs to be matching the stored value.
```

File: docs/modules/ROOT/pages/contribute.adoc [hunk 16]
```diff
@@ -0,0 +1,48 @@
...
+Each TMail sub-project have a dedicated Github project where we can open issues and propose pull requests.
+
+== What can we contribute?
+
+Most people tend to think that contributions need to be code, but we believe here that anyone can contribute. Yes, *anyone*!
+
+ - You can share your experience with TMail on social medias or via a blog post, this would greatly help us!
+ - Reporting bugs, proposing features, giving a feedback is a valuable contribution too!
+ - Documentation is very helpful. Hopefully the existing one is already clear enough, but who knows?
+ - We gadly accept bug fixes and additional
```
Comment (chibenwa) [inline docs/modules/ROOT/pages/contribute.adoc:18]:
```suggestion
 - We gadly accept bug fixes and additional features.
```

File: docs/modules/ROOT/pages/contribute.adoc [hunk 17]
```diff
@@ -0,0 +1,46 @@
...
+
+- link:https://github.com/linagora/tmail-flutter/issues[ISSUES]
+- link:https://github.com/linagora/tmail-flutter/pulls[PRS]
+
+== Coding standards
+
+We encourage the following practices to help your code contributions to be integrated:
+
+ - We encourage to discuss your proposals in an issue before going straight to coding. We likely have valuable insight...
+ - Follow the checkstyle code formatting, maven/gradle ordering rules. If compilation pass, you are good!
```
[the remark this answers is not in the document]
Comment (chibenwa) [reply docs/modules/ROOT/pages/contribute.adoc:41]: Mobile

File: docs/modules/ROOT/pages/ethic.adoc [hunk 18]
```diff
@@ -0,0 +1,123 @@
...
+image::open-source.png[Open Source]
+
+As such, we are only writing OpenSource software. Yes. Only OpenSource. All of our main components are licensed with the
+**AGPL** Linagora license.
+
+This is a list of our repositories licensed with the **AGPL** license:
+
+- link:https://github.com/linagora/tmail-backend/[TMail backend]
+- link:https://github.com/linagora/tmail-flutter[TMail mobile]
+- link:https://github.com/linagora/esn-frontend-inbox[OpenPaaS INBOX]
```
[the remark this answers is not in the document]
Comment (chibenwa) [reply docs/modules/ROOT/pages/ethic.adoc:41]: For now yes

**Review by Arsnael — APPROVED** [review]

**Pull request merged**